src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Policy/GetAzurePolicyAssignment.cs

```diff
@@ -18,60 +18,60 @@ namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
+        [Parameter(ParameterSetName = PolicyCmdletBase.EffectiveFilterParameterSet, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = PolicyHelpStrings.GetPolicyAssignmentScopeHelp)]
+        [Parameter(ParameterSetName = PolicyCmdletBase.AllFilterParameterSet, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = PolicyHelpStrings.GetPolicyAssignmentScopeHelp)]
+        [Parameter(ParameterSetName = PolicyCmdletBase.PolicyDefinitionIdFilterParameterSet, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = PolicyHelpStrings.GetPolicyAssignmentScopeHelp)]
         [ValidateNotNullOrEmpty]
         public string Scope { get; set; }
 
         /// <summary>
         /// Gets or sets the policy assignment id parameter
         /// </summary>
         [Alias("ResourceId")]
-        [Parameter(ParameterSetName = GetAzurePolicyAssignmentCmdlet.PolicyAssignmentIdParameterSet, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The fully qualified policy assignment Id, including the subscription. e.g. /subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}")]
+        [Parameter(ParameterSetName = PolicyCmdletBase.IdParameterSet, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = PolicyHelpStrings.GetPolicyAssignmentIdHelp)]
         [ValidateNotNullOrEmpty]
         public string Id { get; set; }
 
         /// <summary>
         /// Gets or sets the policy assignment policy definition id parameter
         /// </summary>
-        [Parameter(ParameterSetName = GetAzurePolicyAssignmentCmdlet.PolicyAssignmentIdParameterSet, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The fully qualified policy assignment Id, including the subscription. e.g. /subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}")]
-        [Parameter(ParameterSetName = GetAzurePolicyAssignmentCmdlet.PolicyAssignmentNameParameterSet, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The fully qualified policy assignment Id, including the subscription. e.g. /subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}")]
+        [Parameter(ParameterSetName = PolicyCmdletBase.PolicyDefinitionIdFilterParameterSet, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = PolicyHelpStrings.GetPolicyDefinitionFilterHelp)]
         [ValidateNotNullOrEmpty]
         public string PolicyDefinitionId { get; set; }
 
+        /// <summary>
+        /// Gets or sets the all switch
+        /// </summary>
+        [Parameter(ParameterSetName = PolicyCmdletBase.AllFilterParameterSet, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = PolicyHelpStrings.GetPolicyAssignmentAllFilterHelp)]
+        public SwitchParameter All { get; set; }
```

Comment:
I think it would be confusing, since the default without a filter is -Effective which gives less than -All. Basically, that approach would give you more items when you apply a filter than when you don't. The default can't be changed without a big breaking change.

Reply:
Just to make sure I understand, the current behavior without either of these switch parameter is "filter by effective", and the cmdlet doesn't return all policy assignments? If so, is there a difference between `Get-AzureRmPolicyAssignment` and `Get-AzureRmPolicyAssignment -Effective`?

Reply:
That's correct, so there's no difference. I included -Effective so there is a name for what happens by default.

Reply:
I think then rather exposing this parameter that has no effect, we rename the `-All` parameter to describe what type of policy assignments are being returned (if they have a name) in addition to the effective ones returned by default. The parameter would then be renamed to be `-IncludeFoo`, where Foo is the name of these additional assignments.

Reply:
OK. Unfortunately there isn't really a name for that set. I'll remove -Effective and try to think of a better name for -All. So far, the best I have is -IncludeDescendents.